### DISCUSSION TITLE
How to use a sub query in the from clause in the dsl

### DISCUSSION THREAD

**Discussion by msarvar** [diesel-rs/diesel#3513] [Q&A] [answered]
I'm having a hard time figuring out how to build a dsl query to generate equivalent of the following sql.
```
SELECT * FROM (SELECT ROW_NUMBER() OVER (ORDER BY timestamp), * FROM t1) WHERE t1.colA = b;
```

It is similar to https://github.com/diesel-rs/diesel/discussions/3468 discussion, but looks author found a way around changing from clause. I don't think it is an option in my case.

**Comment by weiznich** [comment, marked as answer]
#3468 is completely unrelated. Otherwise you failed to find something about that, as that's just not supported by diesels built-in dsl yet. We are open for contributions on how to design and implement such a feature. If you have ideas please submit a new idea thread outlining the details.

As for how to solve this issue: I see two potential solutions:

* Use [`diesel::sql_query`](https://docs.diesel.rs/master/diesel/fn.sql_query.html)
* Build your own dsl extension. See the "Extending Diesel" guide on our webpage for an example.

**Reply by msarvar** [reply]
Thanks for responding. I will look into making my own dsl otherwise I would have rewrite a lot of pre-existing dsl as raw sql.
